crates/noirc_frontend/src/node_interner.rs

```diff
@@ -149,6 +150,8 @@ pub struct NodeInterner {
     // methods from impls to the type.
     structs: HashMap<StructId, Rc<RefCell<StructType>>>,
 
+    global_constants: HashMap<Ident, StmtId>, // NOTE: currently only used for checking repeat global consts and
```

Comment:
It seems wrong to map from Ident to StmtId since that wouldn't allow for globals with the same name in different modules. Perhaps we should map from ModuleDefId or something else instead